src/emqtt.erl

```diff
@@ -1027,15 +1002,6 @@ code_change(_Vsn, State, Data, _Extra) ->
 %% Internal functions
 %%--------------------------------------------------------------------
 
-should_ping(ConnMod, Sock) ->
-    case ConnMod:getstat(Sock, [send_oct]) of
-        {ok, [{send_oct, Val}]} ->
-            OldVal = get(send_oct), put(send_oct, Val),
-            OldVal == undefined orelse OldVal == Val;
```

Comment:
since PING is a one-byte request the correct fix should be to check if `OldVal < Val -1` ?

Reply:
PINGREQ has more than one byte. If we insist on subtracting the length of the PINGREQ ourselves, it is okay, but the force_ping mechanism is useless. It is simplest to send PINGREQ at a fixed time.